lib/smalltalk.js

```diff
@@ -8,11 +8,15 @@ const keyDown = currify(keyDown_);
 
 const remove = bind(removeEl, '.smalltalk');
 
-const BUTTON_OK = ['OK'];
-const BUTTON_OK_CANCEL = ['OK', 'Cancel'];
+const BUTTON_OK = {'ok':'OK'};
+const BUTTON_OK_CANCEL = {'ok':'OK', 'cancel':'Cancel'};
 
-exports.alert = (title, msg) => {
-    return showDialog(title, msg, '', BUTTON_OK, {cancel: false});
+exports.alert = (title, msg, options) => {
+    const buttons = getButtons(options) || BUTTON_OK;
+    if (getButtons(options) && options.cancel != false) {
+      options.cancel = true;
+    }
```

Comment:
What is this check used for? It is better to not change objects that come from the outside.

Reply:
Because when creating the Promise into `ShowDialog` function it does this : `const noCancel = options && !options.cancel;` and if I pass an options object this became `true` and the cancel callback doesn't work anymore. I think that override the `cancel` option only if not true could be a good idea..
Any other idea how to fix this?

Reply:
In the next line you can see this:

```js
cancel(noCancel ? empty : reject);
```
So this is by design if you press cancel - do nothing, if you press ok - do something, that is it.

Reply:
So if `options` is `null` or `{}` should be the same, but is not : 
 - `null` -> `noCancel` becomes true -> `reject` (trigger the onCancel callback)
 - `{}` -> `noCancel` becomes false -> `empty` (no trigger)

Maybe I'm wrong but thinking on it `noCancel` should be `true` only when `options.cancel` is `false` (also because into REAME is wrote that the default value is `options.cancel = true`) .
So I think we can change the check to : 
```
const noCancel = options && options.cancel == false 
```

Reply:
I understood what I you talking about, yes you are right :)

Reply:
Fixed with 47a7311, landed in [v3.1.1](https://github.com/coderaiser/smalltalk/releases/tag/v3.1.1)